include/dxc/Support/microcom.h

```diff
@@ -104,7 +109,8 @@ inline T *CreateOnMalloc(IMalloc * pMalloc, Args&&... args) {
       ULONG result = InterlockedDecrement(&m_dwRef); \
       if (result == 0) { \
         CComPtr<IMalloc> pTmp(m_pMalloc); \
-        DxcThreadMalloc M(pTmp); delete this; \
```

Comment:
I think you still need to set up the DxcThreadMalloc, for the cases where sub-objects within this need to have it set up. It's implied that an object with its own IMalloc will use it for internal objects, even if those are new/delete-based.

Reply:
I added it back in. Also fixed a class that does not go through the usual microcom construction that sets the IMalloc member. All tests are passing now.